tinygrad/codegen/uopgraph.py

```diff
@@ -562,6 +565,17 @@ def push(u:UOp):
       if x.op is UOps.DEFINE_ACC:
         idx = min([self._uops.index(l) for l in x.src if l.op is UOps.RANGE])
         self._uops.insert(idx, x)
+      elif x.op is UOps.STORE and len(x.src) == 4 and x.src[-1].op is UOps.IF:
```

Comment:
If possible, it's much better to do things as graph rewrite rules than here. Does this have to be done in the linearizer?